src/api/helpers/formatEntityData.ts

```diff
@@ -99,6 +99,82 @@ export function getWorkBasicInfo(work: Record<string, unknown> | null | undefine
 		};
 }
 
+/**
+ * A function to extract structured author credit details from an Edition ORM entity.
+ * @function
+ * @param {object} edition - an edition ORM entity
+ * @returns {object|null} an object containing the author credit data, or null if not available.
+ *
+ * @example
+ * 		const edition = await orm.func.entity.getEntity(orm, 'Edition', bbid, relations);
+ *		filterAuthorCredit(edition);
+ *		/* => {
+ *			"authorCount": 1,
+ *			"id": 135,
+ *			"names": [
+ *				{
+ *					"authorBBID": "e5c4e68b-bfce-4c77-9ca2-0f0a2d4d09f0",
+ *					"authorCreditID": 135,
+ *					"name": "J. K. Rowling"
+ *				}
+ *			]
+ *		}
+ */
+
+export function filterAuthorCredit(edition: any) {
```

Comment:
In terms of naming conventions, "filter" suggests that we are manipulating the results to filter out some items using some comparison or other.
I don't think this is what this function does, so I suggest something like `getAuthorCredits`